src/lib/hooks.js

```diff
@@ -59,21 +59,17 @@ const api = {
 		});
 	},
 
-	setDepartment(departmentValue) {
-		let department = null;
+	setDepartment(value) {
 		const { config: { departments = [] } } = store.state;
 
-		const deptExists = departments.some((dept) => dept._id === departmentValue);
-		if (deptExists) {
-			department = departmentValue;
-		}
+		const dept = departments.find((dep) => dep._id === value || dep.name === value);
+		const department = (dept && dept._id) || '';
 
 		updateIframeGuestData({ department });
 	},
 
 	clearDepartment() {
-		const { iframe: { guest, ...iframe } } = store.state;
-		store.setState({ iframe: { ...iframe, guest: { ...guest, department: null } } });
+		updateIframeGuestData({ department: '' });
```

Comment:
Why an empty string instead of `null`?

Reply:
Yeah, I know about that.. It's another case related to the SDK methods, We'll need to work on the SDK and then we'll be able to pass `null` instead of `string`.